core/embed/projects/prodtest/main.c

```diff
@@ -77,6 +77,12 @@
 
 #include "memzero.h"
 
+#ifdef USE_POWERCTL
+#include <sys/powerctl.h>
+#include "../../sys/powerctl/npm1300/npm1300.h"
```

Comment:
Yes, I agree it looks a bit ugly. I like the idea of the prodtest code being partially integrated into the driver itself. However, this would require some changes to the prodtest philosophy. Let’s discuss it further. For now, I would leave this detail as it is.

Reply:
i created an issue for prodtest redesign, see https://github.com/trezor/trezor-firmware/issues/4430. lets collect ideas for improvement there.